flink-formats/flink-avro-confluent-registry/src/main/java/org/apache/flink/formats/avro/registry/confluent/RegistryAvroFormatFactory.java

```diff
@@ -136,6 +160,31 @@ public String factoryIdentifier() {
     public Set<ConfigOption<?>> optionalOptions() {
         Set<ConfigOption<?>> options = new HashSet<>();
         options.add(SCHEMA_REGISTRY_SUBJECT);
+        options.add(SCHEMA_REGISTRY_SSL_KEYSTORE_LOCATION);
+        options.add(SCHEMA_REGISTRY_SSL_KEYSTORE_PASSWORD);
+        options.add(SCHEMA_REGISTRY_SSL_TRUSTSTORE_LOCATION);
+        options.add(SCHEMA_REGISTRY_SSL_TRUSTSTORE_PASSWORD);
+        options.add(SCHEMA_REGISTRY_BASIC_AUTH_CREDENTIALS_SOURCE);
+        options.add(SCHEMA_REGISTRY_BASIC_AUTH_USER_INFO);
         return options;
     }
+
+    private Map<String, ?> buildOptionalPropertiesMap(ReadableConfig formatOptions) {
+        Map<String, String> optionalPropertiesMap = new HashMap<>();
+
+        List<ConfigOption<String>> configOptions = new ArrayList<>();
+        configOptions.add(SCHEMA_REGISTRY_SSL_KEYSTORE_LOCATION);
+        configOptions.add(SCHEMA_REGISTRY_SSL_KEYSTORE_PASSWORD);
+        configOptions.add(SCHEMA_REGISTRY_SSL_TRUSTSTORE_LOCATION);
+        configOptions.add(SCHEMA_REGISTRY_SSL_TRUSTSTORE_PASSWORD);
+        configOptions.add(SCHEMA_REGISTRY_BASIC_AUTH_CREDENTIALS_SOURCE);
+        configOptions.add(SCHEMA_REGISTRY_BASIC_AUTH_USER_INFO);
```

Comment:
i dont think this is sufficient, for example schema registry supports bearer token too
https://github.com/confluentinc/schema-registry/blob/master/client/src/main/java/io/confluent/kafka/schemaregistry/client/SchemaRegistryClientConfig.java#L34

I think a better solution is to pass all properties that start with `schema-registry` prefix to schema registry client so that we dont need be coupled with the implementation of schema registry. this is similar to how table api pass kafka properties to kafka

Reply:
+1 to this, we should not come up with our own properties because this means more maintenance overhead in the long-term when new properties are introduced. A prefix should be sufficient.

Reply:
Yeah I was thinking about this as well but I wasn't sure how that works with the [RegistryAvroOptions](https://github.com/apache/flink/blob/master/flink-formats/flink-avro-confluent-registry/src/main/java/org/apache/flink/formats/avro/registry/confluent/RegistryAvroOptions.java)

To my understanding I have to add one `ConfigOption` for each property?

Reply:
You are right, this is a shortcoming in the current FormatFactory design. In dynamic table factories you have access to the plain `Map<String, String>` for excluding prefixes but not in format factories. We should fix this in a follow up issue. I will create a JIRA ticket for it.

Until then let's do hard coded properties but already in prefixed format. Then we can replace those explicit options with wildcard options in the future.

Reply:
I created https://issues.apache.org/jira/browse/FLINK-22878. But this might take a while because it would affect many components in Flink to solve it properly.